src/hotspot/share/gc/g1/g1HeapSizingPolicy.hpp

```diff
@@ -66,6 +71,8 @@ class G1CollectedHeap;
 // For full collections, we base resize decisions only on Min/MaxHeapFreeRatio.
 //
 class G1HeapSizingPolicy: public CHeapObj<mtGC> {
+  static jlong _uncommit_delay_ms;  // Delay before uncommitting inactive regions
```

Comment:
It's generally better to use `Ticks`/`Timespan` etc for times as they a) have higher precision and b) one does not need to carry the unit in them. Just convert when you need. They are also distinct types so that wrong assignment can not happen (like a value containing seconds to something that ought to contain milliseconds).
One can only round down etc. when needed.

Also, `jlong` should be avoided, as it is originally a "J"ava type, i.e. for Java interfacing (yeah, it has crept in over time for various reasons).

Reply:
Great points. Will update to:

```
Ticks current_ticks = Ticks::now();
Tickspan elapsed = current_ticks - _last_access_ticks;
return elapsed > _uncommit_delay;
```